docs(examples): simplify docs using new layout methods

joshka opened the pull request:
Use the new `Layout::horizontal` and `vertical` constructors and
`Rect::split_array` through all the examples.

joshka commented:
Removed the helper methods. They don't make things particularly clearer and have problems with polluting the imports. They're tough to put in the prelude due to the common names (min/max/length), which overall suggests that we're better without them. Diff when removed was 254 lines added, 258 removed, so the benefit on LoC is negligible.

joshka commented:
Rebased on the current version of main with Rect:split()

joshka commented:
Regarding testing this, the easiest way is to run `makers run-examples` which runs each example one after the other. All still work fine.

joshka commented:
Rebased now that #729 is merged.

Valentin271 reviewed (approved):
Looks good, I can confirm all examples are working and layouts are visually ok.

kdheepak commented:
I think it is a little confusing that we have 

```
let rects = layout.split(area);
```

and we have

```
let [rect1, rect2] = area.split(layout);
```

Is there a reason we couldn't always use the second form?
If there is a good reason for choosing the first form, I think at the very least it should be documented on the website.

joshka commented:
> Is there a reason we couldn't always use the second form? If there is a good reason for choosing the first form, I think at the very least it should be documented on the website.

The area.split(layout) form requires knowing the number of rects that will be returned beforehand. The layout.split(area) form does not. The places where I've left that are places where there are a dynamic number of areas in which to split to.

I also left a couple of more complex examples alone (IIRC it was in the old demo).

You're right - this could do with some docs in `Layout::split`

kdheepak commented on examples/block.rs:
```diff
@@ -103,20 +103,14 @@ fn ui(frame: &mut Frame) {
+    let main_layout = Layout::vertical([Constraint::Length(1), Constraint::Min(0)]);
+    let block_layout = &Layout::vertical([Constraint::Max(4); 9]);
+    let [title_area, main_area] = area.split(&main_layout);
+    let main_areas = block_layout
+        .split(main_area)
```
Just for my understanding, here even though we know that the number of `Rect`s after the split has to be `2`, the compiler is not able to figure it out at compile time so we have to use `block_layout.split(main_area)`?

Presumably the following would work?

```
main_area.split::<2>(block_layout)
```

joshka replied:
yep
